bot/exts/help_channels/_cog.py

```diff
@@ -108,11 +108,14 @@ async def rename_help_post(self, ctx: commands.Context, *, title: str) -> None:
             # Silently fail in channels other than help posts
             return
 
-        if not await commands.has_any_role(constants.Roles.helpers).predicate(ctx):
-            # Silently fail for non-helpers
-            return
-
-        await ctx.channel.edit(name=title)
+        # First check that the user is the OP as `has_any_role` will only return True
+        # or raise an exception instead of False
+
+        if (
+            ctx.author.id == ctx.channel.owner_id
+            or await commands.has_any_role(constants.Roles.helpers).predicate(ctx)
```

Comment:
Perhaps we could use [this](https://github.com/python-discord/bot/blob/main/bot/utils/checks.py#L97-L107) utility function? Suggested by @TizzySaurus 
```suggestion
            or has_any_role_check(ctx, constants.Roles.helpers)
```
`has_any_role_check` would need to be imported `from bot.utils.checks import has_any_role_check`

Reply:
If that solves the "The implementation is a bit fragile as it depends on short-circuiting ..." problem, then I support it.